### ISSUE TITLE
 Azure IoT Device Workbench leads to VS Code disconnecting from WSL after 1.39 update

### ISSUE THREAD

**Issue by ahernandez-allegion** [microsoft/vscode-remote-release#1646]
Issue Type: <b>Bug</b>

After the automatic update to 1.39, the WSL extension is disconnecting from WSL. This happens after I manually connect, run `code .` from WSL, or use "Open Recent". It seems to connect briefly for about 5 seconds then disconnects and gives me this pop-up:

"Cannot reconnect. Please reload the window".

It gives me two options: Reload window or Cancel. If I Reload window, it just happens again.

I've been working in VS Code with WSL perfectly fine today until after the update.

Version: 1.39.0 (user setup)
Commit: 9df03c6d6ce97c6645c5846f6dfa2a6a7d276515
Date: 2019-10-09T06:58:04.439Z
Electron: 4.2.10
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Windows_NT x64 10.0.17763

Extension version: 0.39.9
VS Code version: Code 1.39.0 (9df03c6d6ce97c6645c5846f6dfa2a6a7d276515, 2019-10-09T06:58:04.439Z)
OS version: Windows_NT x64 10.0.17763
Remote OS version: Linux x64 4.4.0-17763-Microsoft

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Xeon(R) E-2176M  CPU @ 2.70GHz (12 x 2712)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>oop_rasterization: disabled_off<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.78GB (6.22GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL|
|OS|Linux x64 4.4.0-17763-Microsoft|
|CPUs|Intel(R) Xeon(R) E-2176M  CPU @ 2.70GHz (12 x 2712)|
|Memory (System)|15.78GB (6.22GB free)|
|VM|0%|
</details>

**Comment by ahernandez-allegion** [comment]
Seems to be the same issue here: https://github.com/microsoft/vscode-remote-release/issues/1637

**Comment by benedikt-budig** [comment]
I'm experiencing the same problem with version 1.39.0.

**Comment by vianchen** [comment]
This problem persists in 1.39.1. Extension output:

![Snipaste_2019-10-12_00-06-04](https://user-images.githubusercontent.com/23373137/66666916-548ee580-ec84-11e9-927a-50976cd67c9a.png)

Version info:

![Snipaste_2019-10-12_00-08-22](https://user-images.githubusercontent.com/23373137/66666956-6e302d00-ec84-11e9-9770-174739b34871.png)

**Comment by aeschli** [comment]
Can you open a WSL shell and see what's at `~/.vscode-server/bin/88f15d17dca836346e787762685a40bb5cce75a8`.  You should see `server.sh` and other files such as `package.json`.

You can also try to close all VSCode windows and run `rm -r ~/.vscode-server/bin/88f15d17dca836346e787762685a40bb5cce75a8`. It will be installed again on the next start.
Let me know if this reproduces the problem.

**Comment by ahernandez-allegion** [comment]
```~/.vscode-server/bin/88f15d17dca836346e787762685a40bb5cce75a8$ ls
LICENSE  bin  extensions  node  node_modules  out  package.json  product.json  server.sh```

Running `rm -r ~/.vscode-server/bin/88f15d17dca836346e787762685a40bb5cce75a8` and reloading VS Code resulted in the same problem. Re-verified that the above files still exist.

```
$ cat package.json
{
  "name": "Code",
  "version": "1.39.1",
  "dependencies": {
    "@microsoft/applicationinsights-web": "^2.1.1",
    "applicationinsights": "1.0.8",
    "chokidar": "3.1.0",
    "cookie": "^0.4.0",
    "graceful-fs": "4.1.11",
    "http-proxy-agent": "^2.1.0",
    "https-proxy-agent": "^2.2.1",
    "iconv-lite": "0.5.0",
    "jschardet": "1.6.0",
    "native-watchdog": "1.2.0",
    "node-pty": "0.9.0-beta19",
    "nsfw": "1.2.5",
    "onigasm-umd": "^2.2.2",
    "semver-umd": "^5.5.3",
    "spdlog": "^0.9.0",
    "vscode-minimist": "^1.2.1",
    "vscode-proxy-agent": "0.4.0",
    "vscode-ripgrep": "^1.5.7",
    "vscode-textmate": "^4.2.2",
    "vsda": "https://github.com/microsoft/vsda.git#45edd3db96a5e308a5cfc817d1ee42f6d96e1659",
    "xterm": "4.1.0-beta8",
    "xterm-addon-search": "0.2.0",
    "xterm-addon-web-links": "0.2.0",
    "yauzl": "^2.9.2",
    "yazl": "^2.4.3"
  },
  "optionalDependencies": {
    "vscode-windows-ca-certs": "0.1.0",
    "vscode-windows-registry": "1.0.2"
  }
}

$ cat server.sh

#!/usr/bin/env sh
#
# Copyright (c) Microsoft Corporation. All rights reserved.
#

case "$1" in
        --inspect*) INSPECT="$1"; shift;;
esac

ROOT="$(dirname "$0")"

"$ROOT/node" ${INSPECT:-} "$ROOT/out/vs/server/main.js" "$@"
```

**Comment by Coyote139** [comment]
A bit radical, but it solved this issue, I deleted the content of the folder: "C:\Users\yourname\AppData\Roaming\Code" ...

**Comment by ahernandez-allegion** [comment]
> A bit radical, but it solved this issue, I deleted the content of the folder: "C:\Users\yourname\AppData\Roaming\Code" ...

This didn't work for me.

**Comment by K1rdro** [comment]
I'm having the same issue on 1.39.1 as well.

**Comment by aeschli** [comment]
@ahernandez-allegion Can you paste the content of the WSL log when starting up?

**Comment by ahernandez-allegion** [comment]
@aeschli I'm attempting to follow the instructions here: 

https://github.com/Microsoft/WSL/blob/master/CONTRIBUTING.md#8-detailed-logs

but I'm not getting any output. Do you have some instructions I could follow?

**Comment by alzadude** [comment]
Hi there, I'm also affected by this, via the referenced issue https://github.com/BetterThanTomorrow/calva/issues/357.

I'm trying to workaround it by downgrading to an earlier VSCode version, e.g. 1.38.1, 1.37.1, but when I try this the issue is still occurring. I'm also deleting `C:\Users\yourname\AppData\Roaming\Code` on the Windows side, and `~/.vscode-server` on the WSL side when downgrading.

Does anyone have any idea why this is not having the desired effect?

**Comment by aeschli** [comment]
@ahernandez-allegion I meant the Remote-WSL extension log:
![image](https://user-images.githubusercontent.com/6461412/67188363-71e25300-f3ec-11e9-99fc-d63307926623.png)

I'd like to understand, does WSL hang or is it the VSCode Remote server thathas shut down or is hanging? Can you open a WSL shell and see with ps -ax` what's happening?
Does it reproduce with all extensions disabled (except Remote-WSL)?

**Comment by ahernandez-allegion** [comment]
@aeschli 

If I disable all the extensions except Remote-WSL, it works! So I will have to work on isolating which extension is interfering.

Here's the log output. I'm only able to grab it just before it crashes. Once it crashes I cannot access the log. I'm also not able to access the URL it says can be used to debug connection issues.

[2019-10-23 19:02:48.674] Starting VS Code Server inside WSL (default distro)
[2019-10-23 19:02:48.674] Extension version: 0.39.9, Windows build: 17763. Multi distro support: disabled. WSL path support: enabled
[2019-10-23 19:02:48.876] Probing if server is already installed: C:\WINDOWS\System32\wsl.exe  -e sh -c "[ -d ~/.vscode-server/bin/6ab598523be7a800d7f3eb4d92d7ab9a66069390 ] && echo found || ([ -f /etc/alpine-release ] && echo alpine || echo x64)"
[2019-10-23 19:02:48.938] Server install found in WSL
[2019-10-23 19:02:48.939] Launching C:\WINDOWS\System32\wsl.exe sh -c '"$VSCODE_WSL_EXT_LOCATION/scripts/wslServer.sh" 6ab598523be7a800d7f3eb4d92d7ab9a66069390 stable .vscode-server 0  --disable-telemetry' in c:\Users\c-ahernandez\.vscode\extensions\ms-vscode-remote.remote-wsl-0.39.9
[2019-10-23 19:02:49.174] Setting up server environment: Looking for /home/c-ahernandez/.vscode-server/server-env-setup. Not found.
[2019-10-23 19:02:49.174] WSL version: 4.4.0-17763-Microsoft 
[2019-10-23 19:02:49.174] Updating server...
[2019-10-23 19:02:49.174] Starting server: /home/c-ahernandez/.vscode-server/bin/6ab598523be7a800d7f3eb4d92d7ab9a66069390/server.sh  --port=0 --fileWatcherPolling=0 --disable-telemetry
[2019-10-23 19:02:49.776] 
[2019-10-23 19:02:49.776] 
[2019-10-23 19:02:49.776] *
[2019-10-23 19:02:49.776] * Visual Studio Code Server
[2019-10-23 19:02:49.776] *
[2019-10-23 19:02:49.776] * Reminder: You may only use this software with Visual Studio family products,
[2019-10-23 19:02:49.776] * as described in the license https://aka.ms/vscode-remote/license
[2019-10-23 19:02:49.776] *
[2019-10-23 19:02:49.776] 
[2019-10-23 19:02:49.776] 
[2019-10-23 19:02:49.776] IP Address: 172.18.46.81
[2019-10-23 19:02:49.776] IP Address: 169.254.163.141
[2019-10-23 19:02:49.776] IP Address: 10.44.135.231
[2019-10-23 19:02:49.776] Extension host agent listening on 63660
[2019-10-23 19:02:49.776] 
[2019-10-23 19:02:49.776] 
[2019-10-23 19:02:49.776] [12:02:49] Extension host agent started.
[2019-10-23 19:02:49.776] WSL resolver response: 127.0.0.1:63660
[2019-10-23 19:02:49.776] To debug connection issues, open a local browser on http://127.0.0.1:63660/version

**Comment by ahernandez-allegion** [comment]
@aeschli 

I was able to figure out which extension was causing the problem: **Azure IoT Device Workbench**

When I disable that extension, Remote-WSL works as expected.

![image](https://user-images.githubusercontent.com/49734333/67427216-ef4ac680-f598-11e9-8eeb-4e54fbf14700.png)

**Comment by ahernandez-allegion** [comment]
> Hi there, I'm also affected by this, via the referenced issue [BetterThanTomorrow/calva#357](https://github.com/BetterThanTomorrow/calva/issues/357).
> 
> I'm trying to workaround it by downgrading to an earlier VSCode version, e.g. 1.38.1, 1.37.1, but when I try this the issue is still occurring. I'm also deleting `C:\Users\yourname\AppData\Roaming\Code` on the Windows side, and `~/.vscode-server` on the WSL side when downgrading.
> 
> Does anyone have any idea why this is not having the desired effect?

@alzadude Try disabling your extensions and see if that works. If it does, you'll need to isolate which extension might be causing a problem. For me it was the **Azure IoT Device Workbench** extension.

**Comment by aeschli** [comment]
@ahernandez-allegion Great find! I tried to reproduce but wasn't successful yet. Would you mind filing an issue against https://github.com/Microsoft/vscode-azure-iot-toolkit describing your settings for the extension?

**Comment by alzadude** [comment]
Hi @ahernandez-allegion,

The only extensions I have installed in VS Code are:

- Remote WSL 0.39.9 (local)
- Calva 2.0.52 (local and WSL)

Calva is where I where I experience the issue (in combination with Figwheel Main), and I need those things to work on my project so I can't disable it 🙂

**Comment by K1rdro** [comment]
@ahernandez-allegion 
Azure IoT Device Workbench was the offender in my case as well, thanks for finding that!

**Comment by alzadude** [comment]
> Hi @ahernandez-allegion,
> 
> The only extensions I have installed in VS Code are:
> 
> * Remote WSL 0.39.9 (local)
> * Calva 2.0.52 (local and WSL)
> 
> Calva is where I where I experience the issue (in combination with Figwheel Main), and I need those things to work on my project so I can't disable it 🙂

@alexandrudima @aeschli I noticed the issue title was changed to be specific about the IoT extension. However, for me this is not the cause because I don't have that installed and yet I am still getting the disconnects (see above for my extension and other info). 

Should I create a new issue? I originally raised an issue in the downstream Calva extension project, but after some triage the Calva developers determined that it was an upstream VSCode issue: https://github.com/BetterThanTomorrow/calva/issues/357
